lib/browser/browser.ts

```diff
@@ -15,168 +15,181 @@ import {eventTargetPatch} from './event-target';
 import {propertyDescriptorPatch} from './property-descriptor';
 import {registerElementPatch} from './register-element';
 
-const set = 'set';
-const clear = 'clear';
-const blockingMethods = ['alert', 'prompt', 'confirm'];
-const _global: any =
-    typeof window !== 'undefined' && window || typeof self !== 'undefined' && self || global;
-
-patchTimer(_global, set, clear, 'Timeout');
-patchTimer(_global, set, clear, 'Interval');
-patchTimer(_global, set, clear, 'Immediate');
-patchTimer(_global, 'request', 'cancel', 'AnimationFrame');
-patchTimer(_global, 'mozRequest', 'mozCancel', 'AnimationFrame');
-patchTimer(_global, 'webkitRequest', 'webkitCancel', 'AnimationFrame');
-
-for (let i = 0; i < blockingMethods.length; i++) {
-  const name = blockingMethods[i];
-  patchMethod(_global, name, (delegate, symbol, name) => {
-    return function(s: any, args: any[]) {
-      return Zone.current.run(delegate, _global, args, name);
-    };
-  });
-}
-
-eventTargetPatch(_global);
-// patch XMLHttpRequestEventTarget's addEventListener/removeEventListener
-const XMLHttpRequestEventTarget = (_global as any)['XMLHttpRequestEventTarget'];
-if (XMLHttpRequestEventTarget && XMLHttpRequestEventTarget.prototype) {
-  patchEventTargetMethods(XMLHttpRequestEventTarget.prototype);
-}
-propertyDescriptorPatch(_global);
-patchClass('MutationObserver');
-patchClass('WebKitMutationObserver');
-patchClass('FileReader');
-propertyPatch();
-registerElementPatch(_global);
-
-// Treat XMLHTTPRequest as a macrotask.
-patchXHR(_global);
-
-const XHR_TASK = zoneSymbol('xhrTask');
-const XHR_SYNC = zoneSymbol('xhrSync');
-const XHR_LISTENER = zoneSymbol('xhrListener');
-const XHR_SCHEDULED = zoneSymbol('xhrScheduled');
-
-interface XHROptions extends TaskData {
-  target: any;
-  args: any[];
-  aborted: boolean;
-}
-
-function patchXHR(window: any) {
-  function findPendingTask(target: any) {
-    const pendingTask: Task = target[XHR_TASK];
-    return pendingTask;
+Zone.__load_patch('timers', (global: any, Zone: ZoneType, api: _ZonePrivate) => {
```

Comment:
I like this one!!!, this is much more clear!!!
We should also patch other non-standard libraries in this way, such as 

- Notification
- MediaQuery
- 3rd party promise 
...